src/util/entity.cpp

```diff
@@ -193,43 +193,37 @@ namespace big::entity
 
 	bool load_ground_at_3dcoord(Vector3& location)
 	{
-		float groundZ;
-		bool done = false;
+		constexpr float max_ground_check	= 1000.f;
+		constexpr int max_attempts			= 300;
+		float ground_z						= location.z;
+		int current_attempts				= 0;
 
-		for (int i = 0; i < 10; i++)
+		while (!MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, max_ground_check, &ground_z, FALSE, FALSE) && current_attempts != max_attempts)
 		{
-			for (int z = 0; z < 1000; z += 25)
+			if (MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, max_ground_check, &ground_z, FALSE, FALSE))
```

Comment:
`MISC::GET_GROUND_Z_FOR_3D_COORD`, is getting called twice per frame.

I suggest the following changes:
```cpp
constexpr int max_attempts = 300;
bool found_ground = false;

int i = 0;
do {
    found_ground = MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, max_ground_check, &ground_z, FALSE, FALSE);
    ++i;
} while(!found_ground && i < max_attempts);
```
It's an oversimplified example but at the least with this we prevent calling the same function twice on the same frame.

Reply:
Oh yeah, i have thought about that but forgot to make the change. Thank you for pointing out.